internal/docstore/firedocstore/example_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firedocstore_test
+
+import (
+	"context"
+	"log"
+
+	"gocloud.dev/gcp"
+	"gocloud.dev/internal/docstore"
+	"gocloud.dev/internal/docstore/firedocstore"
+)
+
+func ExampleOpenCollection() {
+	// This example is used in https://gocloud.dev/howto/docstore.
+
+	// Variables set up elsewhere:
+	ctx := context.Background()
+
+	creds, err := gcp.DefaultCredentials(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	client, _, err := firedocstore.Dial(ctx, creds.TokenSource)
+	if err != nil {
+		log.Fatal(err)
+	}
+	coll, err := firedocstore.OpenCollection(client, "my-project", "my-collection", "userID", nil)
```

Comment:
Pursuant to the compound ID resolution #1860, I think we want to be joining these components together. Not in scope for this PR, but can you file an issue?

Reply:
#2312